.github/workflows/images.yml

```diff
@@ -16,27 +16,37 @@ on:
       - "v*"
 
 jobs:
-  list:
+  # JOB to run change detection
+  changes:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
     outputs:
-      images: ${{ steps.images-matrix.outputs.images }}
+      images: ${{ steps.filter.outputs.changes }}
     steps:
-      - uses: actions/checkout@v3
-      - id: images-matrix
-        run: |
-          python3 -c '
-          from pathlib import Path
-          import json
-          paths = [
-              str(path.name)
-              for path in Path("implementations").iterdir()
-              if path.is_dir()
-          ]
-          print(f"images={json.dumps(paths)}")
-          ' >> $GITHUB_OUTPUT
+    - uses: dorny/paths-filter@v2
```

Comment:
I saw this action and am OK with using it (#150 is the relevant issue which you may have seen), but I don't want to maintain this duplicate list of implementations. In other words, I'm OK with merging this only if you dynamically build the `filters` mapping below, or else at worst ensure that every single implementation in the directory is guarded so it must appear in that list.